internal/annotations/annotations.go

```diff
@@ -375,3 +379,45 @@ func ExtractUpstreamPolicy(anns map[string]string) (string, bool) {
 	s, ok := anns[kongv1beta1.KongUpstreamPolicyAnnotationKey]
 	return s, ok
 }
+
+// ExtractTLSVerify extracts the tls-verify annotation value.
+func ExtractTLSVerify(anns map[string]string) (value bool, ok bool) {
+	s, ok := anns[AnnotationPrefix+TLSVerifyKey]
+	if !ok {
+		// If the annotation is not present, we consider it not set.
+		return false, false
+	}
+	verify, err := strconv.ParseBool(s)
+	if err != nil {
+		// If the annotation is present but not a valid boolean string, we consider it not set.
+		return false, false
+	}
+	// If the annotation is present and a valid boolean string, we return the value.
+	return verify, true
+}
+
+// ExtractTLSVerifyDepth extracts the tls-verify-depth annotation value.
+func ExtractTLSVerifyDepth(anns map[string]string) (int, bool) {
+	s, ok := anns[AnnotationPrefix+TLSVerifyDepthKey]
+	if !ok {
+		// If the annotation is not present, we consider it not set.
+		return 0, false
+	}
+	depth, err := strconv.Atoi(s)
+	if err != nil {
+		// If the annotation is present but not a valid integer string, we consider it not set.
+		return 0, false
+	}
+	// If the annotation is present and a valid integer string, we return the value.
+	return depth, true
+}
+
+// ExtractCACertificates extracts the ca-certificates secret names from the annotation.
+// It expects a comma-separated list of certificate names.
+func ExtractCACertificates(anns map[string]string) []string {
+	s, ok := anns[AnnotationPrefix+CACertificatesKey]
+	if !ok {
+		return nil
+	}
+	return strings.Split(s, ",")
```

Comment:
I think you should trim whitespaces too. User should be able to write in the annotation `ca1,ca2` or `ca1, ca2`. The same for any annotation that accepts multiple values and whitespaces are not being a valid part of an item ( I think all cases). WDYT?

Reply:
I was following the approach we had with other annotations, but I agree we could improve this. I handled this properly for the new function and will migrate other functions in a separate PR to isolate these changes. PTAL https://github.com/Kong/kubernetes-ingress-controller/pull/6707/commits/74ccc7a96154eb0e42c02a63149b03b2857193c1

Reply:
Great, I did such a change some time ago too as an opportunistic improvement in the PR https://github.com/Kong/kubernetes-ingress-controller/pull/5977